deps/chakrashim/src/jsrtisolateshim.cc

```diff
@@ -670,6 +671,26 @@ void IsolateShim::SetPromiseRejectCallback(v8::PromiseRejectCallback callback) {
                                    reinterpret_cast<void*>(callback));
 }
 
+void IsolateShim::RunMicrotasks() {
+  // Loop until we've handled all the tasks, including the ones
+  // added by these tasks
+  while (!this->microtaskQueue.empty()) {
+    std::vector<MicroTask> batch;
+    std::swap(batch, this->microtaskQueue);
+
+    for (unsigned int i = 0; i < batch.size(); ++i) {
```

Comment:
You might be able to do:

`for (const MicroTask& microTask : batch) {`

which provides a const reference to the item in the vector (and thus doesn't copy it)